packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -2930,6 +2932,17 @@ export default (superClass: Class<Parser>): Class<Parser> =>
         return expr;
       }, state);
 
+      // report error if single type parameter used without trailing comma.
+      if (
+        arrow.node?.type === "ArrowFunctionExpression" &&
```

Comment:
We should only report the error if `this.hasPlugin("jsx")`

Reply:
Maybe we only check it in the tsx file?